src/libsync/syncfilestatustracker.cpp

```diff
@@ -225,6 +227,32 @@ void SyncFileStatusTracker::slotItemCompleted(const SyncFileItem &item)
     if (showErrorInSocketApi(item)) {
         _stateLastSyncItemsWithErrorNew.insert(item._file);
     }
+
+    QString systemFileName = _syncEngine->localPath() + item.destination();
+
+    // the trailing slash for directories must be appended as the filenames coming in
+    // from the plugins have that too. Otherwise the matching entry item is not found
+    // in the plugin.
+    if( item._type == SyncFileItem::Type::Directory ) {
+        systemFileName += QLatin1Char('/');
+    }
+
+    auto status = fileStatus(item.destination());
+    emit fileStatusChanged(systemFileName, status);
+}
+
+void SyncFileStatusTracker::slotItemDiscovered(const SyncFileItem &item)
+{
+    QString systemFileName = _syncEngine->localPath() + item.destination();
+
+    // the trailing slash for directories must be appended as the filenames coming in
+    // from the plugins have that too. Otherwise the matching entry item is not found
+    // in the plugin.
+    if( item._type == SyncFileItem::Type::Directory ) {
+        systemFileName += QLatin1Char('/');
+    }
```

Comment:
Yes you're probably right, some kind of `toShellPath` and `toSyncEnginePath` functions that would add/remove the slash from directories.